org-cyf-itp/tooling/netlify/functions/submission-created.js

```diff
@@ -0,0 +1,107 @@
+  //  2. Generate or find the service account.
+  //  3. Download its credentials as a JSON file.
+  //  4. Fetch the keys listed below - the non-env-var ones are not really secret.
+  //  5. For private_key, this is read from an env var on netlify, and env vars cannot contain newlines, so we put the string "\n" where we need newlines, accordingly, we undo this below.
+  //     The env var was constructed by running: `jq '.private_key' <cyf-syllabus-7ca5140fd0c6.json | tr -d '"'`.
+  "private_key": process.env["GOOGLEAPI_REGISTER_PRIVATE_KEY"].replaceAll("\\n", "\n"),
+  "client_email": "register@cyf-syllabus.iam.gserviceaccount.com",
+  "client_id": "113977060196146055874",
+};
+
+const handler = async (event, context) => {
+  console.log("Got request with body", event.body);
+  let body;
+  try {
+    // TODO: Check for structure
+    body = JSON.parse(event.body);
+  } catch (error) {
+    console.error(`Failed to parse request as valid JSON: ${event.body}: ${error}`);
+    return {statusCode: 400, body: JSON.stringify("Failed to parse body as JSON")};
+  }
+  if (!("payload" in body) || !("data" in body.payload)) {
+    console.error(`Failed to parse request - missing .payload.data: ${event.body}`);
+    return {statusCode: 400, body: JSON.stringify("Failed to parse body as JSON")};
+  }
+  const request = body.payload.data;
+
+  for (const requiredField of ["course", "module", "givenName", "familyName", "email", "day"]) {
+    if (!(requiredField in request)) {
+      const error = `Request missing field ${requiredField}`;
+      console.error(error, request);
+      return {statusCode: 400, body: JSON.stringify(error)};
+    }
+    const type = typeof request[requiredField];
+    if (type !== "string") {
+      const error = `Field ${requiredField} must be of type string but was ${type}`;
+      console.error(error, request);
+      return {statusCode: 400, body: JSON.stringify(error)};
+    }
+  }
+
+  if (!(request.course in COURSE_TO_SPREADSHEET_ID)) {
+    console.error(`Didn't recognise course ${request.course}`);
+    return {statusCode: 404, body: JSON.stringify(`Unknown course: ${request.course}`)};
+  }
+
+  const auth = new Auth.GoogleAuth({
+    credentials: CREDENTIALS,
+    scopes: "https://www.googleapis.com/auth/spreadsheets",
+  });
+
+  const sheets = google.sheets({version: "v4", auth});
+
+  try {
+    const response = await sheets.spreadsheets.values.append({
+      spreadsheetId: COURSE_TO_SPREADSHEET_ID[request.course],
+      // Trust that the user-supplied module exists.
+      // If it's wrong, we'll get a 400 from the API and pass it on to the user without much useful detail.
+      range: request.module,
+      valueInputOption: "RAW",
+      insertDataOption: "INSERT_ROWS",
+      resource: {
+        values: [
+          [request.givenName, request.familyName, request.email, new Date().toISOString(), request.course, request.module, request.day],
```

Comment:
FYI, the reason I included two dates was that I wanted to easily colour the sign in red if the signing date was not within 1 day of the build date of the form, given that the site is rebuilt every night. This was my extremely slapdash/goodenough solution to someone going through and signing in retrospectively or speculatively without having to set up a dictionary of class dates to check against. 

How can you solve this problem?

Reply:
For now, I've just preserved your existing solution :)